.github/workflows/test.yml

```diff
@@ -124,6 +51,15 @@ jobs:
       CC: ${{ matrix.build.cc }}
     steps:
       - uses: actions/checkout@v4
+
+      - if: ${{ matrix.build.flavor == '32' }}
+        run: |
+          sudo dpkg --add-architecture i386
+          sudo apt-get update -y
+          sudo apt-get install -y gcc-13:i386
+          sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-13 100
+          sudo update-alternatives --set gcc /usr/bin/gcc-13
+
       - run: ./.github/scripts/install_deps.sh --test
```

Comment:
This is going to undo nearly everything from the above step because it installs `build-essential` for the native architecture.  That forces gcc-13:i386 to get removed, among other things.

`install_deps.sh` really needs to be architecture aware instead.  Something like

```suggestion
      - run: ./.github/scripts/install_deps.sh --test --arch ${{ matrix.build.flavor }}
```

where `matrix.build.flavor` is either `native` or `i386` (or make the flavor `''` or `'--arch i386'`).

This patch for install_deps.sh should handle that.

```diff
diff --git i/.github/scripts/install_deps.sh w/.github/scripts/install_deps.sh
index 9a782e969..7a3a10e87 100755
--- i/.github/scripts/install_deps.sh
+++ w/.github/scripts/install_deps.sh
@@ -1,18 +1,25 @@
 #!/bin/bash
 
+ARCH=native
+
 while (($# > 0)); do
   case $1 in
   --test) # install test dependencies
     TEST=1
     shift
     ;;
+  --arch)
+    shift
+    ARCH="$1"
+    shift
+    ;;
   esac
 done
 
 os=$(uname -s)
 if [[ $os == Linux ]]; then
   sudo apt-get update
-  sudo apt-get install -y build-essential cmake curl gettext ninja-build
+  sudo apt-get install -y build-essential:$ARCH cmake curl gettext ninja-build
 
   if [[ $CC == clang ]]; then
     DEFAULT_CLANG_VERSION=$(echo |  clang -dM -E - | grep __clang_major | awk '{print $3}')
@@ -30,7 +37,7 @@ if [[ $os == Linux ]]; then
   fi
 
   if [[ -n $TEST ]]; then
-    sudo apt-get install -y locales-all cpanminus attr libattr1-dev gdb
+    sudo apt-get install -y locales-all cpanminus attr libattr1-dev:$ARCH gdb
   fi
 elif [[ $os == Darwin ]]; then
   brew update --quiet
```